lib/msf/core/exploit/remote/smb/server/hash_capture.rb

```diff
@@ -0,0 +1,191 @@
+      # Set challenge for all future server responses
+
+      chall = proc { [datastore['CHALLENGE']].pack('H*') }
+      ntlm_provider.generate_server_challenge(&chall)
+    end
+
+    if datastore['JOHNPWFILE']
+      print_status("JTR hashes will be split into two files depending on the hash format.")
+      print_status("#{build_jtr_file_name(JTR_NTLMV1)} for NTLMv1 hashes.")
+      print_status("#{build_jtr_file_name(JTR_NTLMV2)} for NTLMv2 hashes.")
+      print_line
+    end
+
+    if datastore['CAINPWFILE']
+      print_status("Cain & Abel hashes will be stored at #{File.expand_path(datastore['CAINPWFILE'], Msf::Config.install_root)}")
+      print_line
+    end
+  end
+
+  def report_ntlm_type3(address:, ntlm_type1:, ntlm_type2:, ntlm_type3:)
+    ntlm_message = ntlm_type3
+    hash_type = nil
+
+    user = ntlm_message.user.force_encoding(::Encoding::UTF_16LE).encode(''.encoding)
+    domain = ntlm_message.domain.force_encoding(::Encoding::UTF_16LE).encode(''.encoding)
+    challenge = [ntlm_type2.challenge].pack('Q<')
+    combined_hash = "#{user}::#{domain}"
+
+    case ntlm_message.ntlm_version
+    when :ntlmv1
+      hash_type = 'NTLMv1-SSP'
+      client_hash = "#{bin_to_hex(ntlm_message.lm_response)}:#{bin_to_hex(ntlm_message.ntlm_response)}"
+
+      combined_hash << ":#{client_hash}"
+      combined_hash << ":#{bin_to_hex(challenge)}"
+    when :ntlmv2
+      hash_type = 'NTLMv2-SSP'
+      client_hash = "#{bin_to_hex(ntlm_message.ntlm_response[0...16])}:#{bin_to_hex(ntlm_message.ntlm_response[16..-1])}"
+
+      combined_hash << ":#{bin_to_hex(challenge)}"
+      combined_hash << ":#{client_hash}"
+    end
+
+    return if hash_type.nil?
+
+    # TODO: write method for mapping +major+ and +minor+ OS values to human-readable OS names.
+    # client_os_version = ::NTLM::OSVersion.read(type1_msg.os_version)
+    print_line "[SMB] #{hash_type} Client     : #{address}"
+    # print_line "[SMB] #{hash_type} Client OS  : #{client_os_version}"
+    print_line "[SMB] #{hash_type} Username   : #{domain}\\#{user}"
+    print_line "[SMB] #{hash_type} Hash       : #{combined_hash}"
+    print_line
+
+    jtr_format = ntlm_message.ntlm_version == :ntlmv1 ? JTR_NTLMV1 : JTR_NTLMV2
+
+    if active_db?
+      origin = create_credential_origin_service(
+        {
+          address: address,
+          port: datastore['SRVPORT'],
+          service_name: 'smb',
+          protocol: 'tcp',
+          module_fullname: fullname,
+          workspace_id: myworkspace_id
+        }
+      )
+
+      credential_options = {
+        origin: origin,
+        origin_type: :service,
+        address: address,
+        port: datastore['SRVPORT'],
+        service_name: 'smb',
+        username: user,
+        server_challenge: challenge,
+        client_hash: client_hash,
+        # client_os_version: client_os_version,
+        private_data: combined_hash,
+        private_type: :nonreplayable_hash,
+        module_fullname: fullname,
```

Comment:
I had to add this line to get the JTR type to be set like it was in the SMB capture module prior to these changes.

```suggestion
        private_type: :nonreplayable_hash,
        jtr_format: jtr_format,
        module_fullname: fullname,
```

That ensures that hashes are reported with the more specific JTR subtype:

```
msf6 auxiliary(server/capture/smb) > creds
Credentials
===========

host             origin           service        public     private                                                                                              realm  private_type        JtR Format
----             ------           -------        ------     -------                                                                                              -----  ------------        ----------
192.168.159.128  192.168.159.128  445/tcp (smb)  smcintyre  smcintyre::.:51405c117e04cf3f:c2241057883d9ab58ffc3883b13828cc:01010000000000008028d906 (TRUNCATED)  .      Nonreplayable hash  netntlmv2

msf6 auxiliary(server/capture/smb) >
```

Reply:
Fixed in the last commit 🤞 

```
msf6 exploit(windows/smb/smb_relay) > Credentials
===========

host            origin          service        public  private                                                                                              realm  private_type        JtR Format
----            ------          -------        ------  -------                                                                                              -----  ------------        ----------
192.168.123.28  192.168.123.28  445/tcp (smb)  alan    alan::ADF2:874ed9946d3198ff:0002ecce24e34872773b734860bb69d4:0101000000000000baad7f254c (TRUNCATED)  ADF2   Nonreplayable hash  netntlmv2
```